src/execution/mod.rs

```diff
@@ -65,43 +78,82 @@ where
         };
 
         if let Some(start) = validation_info.start {
-            let result = instance.invoke_func::<(), ()>(start, ());
+            // "start" is not always exported, so we need create a non-API exposed function reference.
+            // Note: function name is not important here, as it is not used in the verification process.
+            let start_fn = FunctionRef {
+                module_name: module_name.to_string(),
+                function_name: "start".to_string(),
+                module_index: 0,
+                function_index: start,
+                exported: false,
+            };
+            let result = instance.invoke::<(), ()>(&start_fn, ());
             result?;
         }
 
         Ok(instance)
     }
 
-    pub fn invoke_named<Param: InteropValueList, Returns: InteropValueList>(
-        &mut self,
-        func_name: &str,
-        param: Param,
-    ) -> Result<Returns, RuntimeError> {
-        // TODO: Optimize this search for better than linear-time. Pre-processing will likely be required
-        let func_idx = self.exports.iter().find_map(|export| {
-            if export.name == func_name {
-                match export.desc {
-                    ExportDesc::FuncIdx(idx) => Some(idx),
-                    _ => None,
-                }
-            } else {
-                None
-            }
-        });
+    pub fn get_fn(
```

Comment:
Please rename this and `get_fn_idx` to `get_function_by_name` and `get_function_by_index`.
Especially `get_fn_idx` is misleading as the name implies that is should return an index.